File: rxui-sample-java/src/test/java/com/artemzin/rxui/sample/java/TestAuthService.java
package com.artemzin.rxui.sample.java;

import android.support.annotation.NonNull;

import io.reactivex.Observable;
import io.reactivex.subjects.PublishSubject;

class TestAuthService implements AuthService {

    final PublishSubject<Object> signIn = PublishSubject.create();

    @NonNull
    @Override
    public Observable<Object> signIn(@NonNull String login, @NonNull String password) {
        return signIn;
    }
}

File: rxui-sample-kotlin/src/test/kotlin/com/artemzin/rxui/sample/kotlin/TestAuthService.kt
package com.artemzin.rxui.sample.kotlin

import com.artemzin.rxui.sample.kotlin.AuthService.Response
import io.reactivex.subjects.PublishSubject

open class TestAuthService : AuthService {

    val signIn = PublishSubject.create<Response>()

    override fun signIn(login: String, password: String) = signIn
}
